tests/integration/dynamic/nodejs-pulumi-config/index.ts

```diff
@@ -10,9 +10,9 @@ class CustomResource extends pulumi.dynamic.Resource {
 
 class DummyResourceProvider implements pulumi.dynamic.ResourceProvider {
   async create (props: any): Promise<pulumi.dynamic.CreateResult> {
-    var config = JSON.parse(process.env.PULUMI_CONFIG)
+    const config = new pulumi.Config("pulumi-nodejs")
```

Comment:
If we want to allow dynamic providers to access config, I think we'll want to allow this usage:

```ts
const config = new pulumi.Config();
```

When no arg is passed to `Config`, the project name will be used by default. I don't think this will work as-is in Node.js, because we're not setting up the project name internally in the SDK. It doesn't work in Python in my testing.

Reply:
Project name is, like config, consistently set via envvars, but again that's not a documented interface from the engine its just how all the language plugins _happen_ to work today. So we could hack upon hacks and set that envvar for dynamic providers as well.

Reply:
We do pass the project to policy packs, so at least it'd be consistent with that.

I do slightly worry about the potential impact to Programs as Components. Right now dynamic providers don't work for MLCs, but we may potentially want to make them work for Programs as Components. In that case, we'd have to plumb the right config and project name through. I suppose we can cross that bridge if/when we need to.